go/parquet/pqarrow/schema.go

```diff
@@ -354,7 +358,15 @@ func fieldToNode(name string, field arrow.Field, props *parquet.WriterProperties
 		return fieldToNode(name, arrow.Field{Name: name, Type: dictType.ValueType, Nullable: field.Nullable, Metadata: field.Metadata},
 			props, arrprops)
 	case arrow.EXTENSION:
-		return nil, xerrors.New("not implemented yet")
+		return fieldToNode(name, arrow.Field{
+			Name: name,
+			Type: field.Type.(arrow.ExtensionType).StorageType(),
+			Nullable: field.Nullable,
+			Metadata: arrow.MetadataFrom(map[string]string{
+				ExtensionTypeKeyName: field.Type.(arrow.ExtensionType).ExtensionName(),
+			}),
```

Comment:
You also need to add `ExtensionMetadataKeyName: field.Type.(arrow.ExtensionType).Serialize()`